commcid.go

```diff
@@ -138,12 +301,44 @@ func validateFilecoinCidSegments(mc FilMultiCodec, mh FilMultiHash, commX []byte
 	return nil
 }
 
+// ConvertDataCommitmentV1V1CIDtoPieceMhCID takes a v1 piece CID and the CommP tree height and produces a
+// piece multihash CID
+//
+// The helpers UnpaddedSizeToV1TreeHeight and Fr32PaddedSizeToV1TreeHeight may help in computing tree height
+func ConvertDataCommitmentV1V1CIDtoPieceMhCID(v1PieceCid cid.Cid, unpaddedDataSize uint64) (cid.Cid, error) {
+	hashDigest, err := CIDToDataCommitmentV1(v1PieceCid)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("Error decoding piece CID v1: %w", err)
+	}
+
+	return DataCommitmentV1ToPieceMhCID(hashDigest, unpaddedDataSize)
+}
+
+// ConvertDataCommitmentV1PieceMhCIDToV1CID takes a piece multihash CID and produces a v1 piece CID along with the
+// size of the unpadded data
+func ConvertDataCommitmentV1PieceMhCIDToV1CID(pieceMhCid cid.Cid) (cid.Cid, uint64, error) {
```

Comment:
Mixing "MH" into the names of things that do not return MHs ( but cid.Cid ) is... nasty.

```suggestion
func PieceCidV1FromV2(pcidV2 cid.Cid) (cid.Cid, uint64, error) {
```